modules/internal/Atomics.chpl

```diff
@@ -430,1359 +310,319 @@ module Atomics {
-    }
-
-    proc const writeThis(x) {
-      x <~> read();
-    }
-  }
-
-  //
-  // For the first cut, we will be making assignment and other normal
-  //  operations illegal.  In addition, we are also punting on
-  //  the default argument intent since it's not obvious what it
-  //  should be.
-  //
+  //
+  // For the first cut, we will be making assignment and other normal
+  // operations illegal.  In addition, we are also punting on
+  // the default argument intent since it's not obvious what it
+  // should be.
+  //
 
   // We need to explicitly define these for all types because the atomic
-  //  types are records and unless explicitly defined, it will resolve
-  //  to the normal record version of the function.  Sigh.
-  inline proc =(ref a:atomicbool, const ref b:atomicbool) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomicbool, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_uint8, const ref b:atomic_uint8) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_uint8, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_uint16, const ref b:atomic_uint16) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_uint16, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_uint32, const ref b:atomic_uint32) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_uint32, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_uint64, const ref b:atomic_uint64) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_uint64, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_int8, const ref b:atomic_int8) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_int8, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_int16, const ref b:atomic_int16) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_int16, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_int32, const ref b:atomic_int32) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_int32, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_int64, const ref b:atomic_int64) {
-    a.write(b.read());
-  }
-  inline proc =(ref a:atomic_int64, b) {
-    compilerError("Cannot directly assign atomic variables");
-  }
-  inline proc =(ref a:atomic_real32, const ref b:atomic_real32) {
+  // types are records and unless explicitly defined, it will resolve
+  // to the normal record version of the function.  Sigh.
+  inline proc =(ref a:AtomicBool, const ref b:AtomicBool) {
     a.write(b.read());
   }
-  inline proc =(ref a:atomic_real32, b) {
+  inline proc =(ref a:AtomicBool, b) {
```

Comment:
I don't understand why we don't support this. Isn't it "obvious" that `myAtomic = 1` will be the same as myAtomic.write(1) ? It seems to me it would be easy to make this `=` overload do that. Might even allow initialization.

Reply:
Supporting atomic initialization and operator overloads are captured elsewhere -- https://github.com/chapel-lang/chapel/issues/5037 and https://github.com/chapel-lang/chapel/issues/8847 (and is also in #9368 as "operator overloads for atomic ops".)